sympytorch/sympy_module.py

```diff
@@ -73,18 +73,22 @@ def __init__(self, *, expr, _memodict, _func_lookup, **kwargs):
             self._value = torch.nn.Parameter(torch.tensor(float(expr)))
             self._torch_func = lambda: self._value
             self._args = ()
-        elif issubclass(expr.func, sympy.UnevaluatedExpr):
-            if len(expr.args) != 1 or not issubclass(expr.args[0].func, sympy.Float):
-                raise ValueError("UnevaluatedExpr should only be used to wrap floats.")
-            self.register_buffer('_value', torch.tensor(float(expr.args[0])))
-            self._torch_func = lambda: self._value
-            self._args = ()
         elif issubclass(expr.func, sympy.Integer):
             # Can get here if expr is one of the Integer special cases,
             # e.g. NegativeOne
             self._value = int(expr)
             self._torch_func = lambda: self._value
             self._args = ()
+        elif issubclass(expr.func, sympy.core.numbers.Half) or issubclass(expr.func, sympy.core.numbers.Rational):
+            self._value = float(expr)
```

Comment:
Can this be `self.register_buffer('_value', torch.tensor(float(expr)))`?

IIRC some `torch` functions really need tensor-valued inputs and float-valued inputs will raise an error, so it'd be best if we converted everything to a tensor first.

(This is analogous to the `UnevaluatedExpr` branch below.)